### DISCUSSION TITLE
Problem setting up passwordless sudo for scripting addition

### DISCUSSION THREAD

**Discussion by ghost** [asmvik/yabai#2162] [Q&A] [answered]
Hey there,

I am a new user trying to make space switching work.
- yabai v7.0.2
- macOS 14.4

I've installed the lastest version and the scripting addition according to the wiki, together with the example config (comment sign before `--load-sa` part is removed). 

Running `yabai -m space --focus 2` results in `cannot focus space due to an error with the scripting-addition.`.

I discovered that my `/tmp/yabai_<username>.err.log` says `sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper`.

Running `sudo yabai --load-sa` also asks for my password.

This I assume means that sudo is not allowing the scripting addition to be executed as root without a password. I used the one-liner from the wiki to create `/etc/sudoers.d/yabai`, it looks like this:

```
root ALL=(root) NOPASSWD: sha256:67c9acb603f51b3d44e798688b4316bc013c9a3486275dfc268d02ed586c5cb5 /usr/local/bin/yabai --load-sa
```

Any pointers? I tried to follow the wiki strictly, everything except this weird sudo error looks good...

Thanks.

**Comment by gegoune** [comment, marked as answer]
First string in that file should be your username, not `root`.

```sh
echo "$(whoami) ALL=(root) NOPASSWD: sha256:$(shasum -a 256 $(which yabai) | cut -d " " -f 1) $(which yabai) --load-sa"
```
If you run that it will print what you need to add to the file, notice `whoami` command, if you did run it before you must have ran it as root. Run it as your regular user instead.

**Reply by ghost** [reply]
Thanks, this solved it! I had prefixed the wiki command with sudo it seems.
